Add release notes for v23.1.11 (#17897)
cockroachdb/docs: src/current/_includes/releases/v23.1/v23.1.11.md

<h3 id="v23-1-11-command-line-changes">Command-line changes</h3>

- Removed the command `\demo recommission` from [`cockroach demo`]({% link v23.1/cockroach-demo.md %}). It had been obsolete and non-functional ever since v20.2. [#108631][#108631]

<h3 id="v23-1-11-db-console-changes">DB Console changes</h3>

- Users without the `VIEWCLUSTERSETTINGS` [permission]({% link v23.1/security-reference/authorization.md %}#supported-privileges) but with the `VIEWACTIVITY` or `VIEWACTIVITYREDACTED` permissions can now see [index recommendations]({% link v23.1/ui-databases-page.md %}#index-recommendations). [#109466][#109466]
- CockroachDB will now show a warning when the time period selected on the [SQL Activity page](../v23.1/ui-overview.html#sql-activity) is older than the oldest data available. [#109468][#109468]

<h3 id="v23-1-11-bug-fixes">Bug fixes</h3>

> - Fixed a buggy TTL descriptor repair by removing it. Previously, upgrading from v22.2.* to 23.1.9 incorrectly removed [TTL storage params]({% link v23.1/row-level-ttl.md %}##ttl-storage-parameters) from tables (visible via [`SHOW CREATE TABLE  <ttl-table>`]({% link v23.1/show-create-table.md %})) while attempting to repair table descriptors. This resulted in the node that attempts to run the TTL [job](../v23.1/show-jobs.html) crashing due to a panic caused by the missing TTL storage parameters. Clusters currently on v22.2.* should **NOT** be upgraded to v23.1.9 and should be upgraded directly to v23.1.10 or later. [#110562][#110562]

Reviewer: Should the failing link be https://www.cockroachlabs.com/docs/stable/show-create#show-the-create-table-statement-for-a-table ?

Author: ah thanks! should be fixed in latest versions

## v23.1.11

> {{ site.data.alerts.callout_danger }}

Reviewer 2: What do you think about using a Note (`site.data.alerts.callout_info`) instead of a Warning?

Reviewer 3: Updated to `callout_info` per Style Guide.

> v23.1.11 is currently available only on select CockroachDB {{ site.data.products.dedicated }} clusters. If you wish to upgrade your {{ site.data.products.dedicated }} cluster to v23.1.11, please contact your account manager.

Reviewer 4: ```suggestion
v23.1.11 is currently available only on select CockroachDB {{ site.data.products.dedicated }} clusters. If you wish to upgrade your {{ site.data.products.dedicated }} cluster to v23.1.11, please submit a [https://support.cockroachlabs.com/hc/en-us/requests/new?](support request).
```

This release is currently not available for CockroachDB {{ site.data.products.serverless }} or {{ site.data.products.core }} customers.
{{ site.data.alerts.end }}

Release Date: September 21, 2023

<h3 id="v23-1-11-sql-language-changes">SQL language changes</h3>

- Added a new syntax to [`SHOW DEFAULT PRIVILEGES`]({% link v23.1/show-default-privileges.md %}), `SHOW DEFAULT PRIVILEGES FOR GRANTEE <grantee>`, that shows the default privileges that a grantee received. [#108285][#108285]
- Admin API [database details]({% link v23.1/cluster-api.md %}) endpoint now returns authoritative range statistics. [#108727][#108727]

> - Added the [cluster setting](../v23.1/cluster-settings.html) `sql.stats.limit_table_size.enabled`, which controls whether or not we enforce the row limit set by `sql.stats.persited_rows.max` in the `system.statement_statistics` and `system.transaction_statistics` tables. [#108912][#108912]

Author: Fixed thank you!

- Optimized the `sql-stats-compaction` [job](../v23.1/show-jobs.html)'s [delete query](../v23.1/delete.html) to avoid a [full scan](../v23.1/sql-tuning-with-explain.html#issue-full-table-scans). This helps avoid a [transaction retry error](../v23.1/common-errors.html#restart-transaction) which can cause the job to fail. [#108987][#108987]
- Fixed an issue where the UI was missing query text and details when looking at the [SQL activity transactions page]({% link v23.1/ui-sql-dashboard.md %}#transactions) if there were more than 500 transactions or statements. The `crdb_internal.statement_activity` table now includes all statements for a transaction that are in the `crdb_internal.transaction_activity` table. [#109479][#109479]
- Added the `VIEWSYSTEMTABLE` [system privilege]({% link v23.1/security-reference/authorization.md %}#supported-privileges). Users with this privilege have [`SELECT`](../v23.1/selection-queries.html) privileges for all tables in the `system` database. [#109525][#109525]
- The `oidvectortypes` [built-in function]({% link v23.1/functions-and-operators.md %}) has been implemented, which can format an `oidvector`. [#109711][#109711]

> - The internal persisted [statistics](../v23.1/cost-based-optimizer.html#table-statistics) table max size check is now done once an hour instead of every 10 minutes. This reduces the risk of serialization errors on the statistics tables. [cockroachdb/cockroach#109707][#109707]

Author: Fixed here and elsewhere, thanks!  Artifact of me looking at the PRs, oops

- Introspection queries will now show the internal `node` user as the owner of tables in [`pg_catalog`]({% link v23.1/pg-catalog.md %}) and [`information_schema`]({% link v23.1/information-schema.md %}). Previously, the owner was shown as `admin`, but that was inaccurate since users with the [`admin` role]({% link v23.1/security-reference/authorization.md %}#admin-role) could not modify these tables in any way. [#109735][#109735]

<h3 id="v23-1-11-operational-changes">Operational changes</h3>

> - Added the `kv.enqueue_in_replicate_queue_on_span_config_update.enabled` [cluster setting.](../v23.1/cluster-settings.html). When set to `true`, [stores](../v23.1/cockroach-start.html#store) in the cluster will enqueue [replicas](../v23.1/architecture/overview.html#architecture-replica) for [replication changes](../v23.1/architecture/replication-layer.html) upon receiving config updates which could affect the replica. This setting is off by default. Enabling this setting speeds up how quickly config-triggered replication changes begin, but adds additional CPU overhead. The overhead scales with the number of leaseholders. [#108812][#108812]

Author: which vs that, I'll never learn ...

- Added a new [cluster setting](../v23.1/cluster-settings.html) named `server.hot_ranges_request.node.timeout`, with a default value of 5 minutes. The setting controls the maximum amount of time that a [hot ranges request]({% link v23.1/ui-hot-ranges-page.md %}) will spend waiting for a node to provide a response. Set it to `0` to disable timeouts. [#109015][#109015]
- Span stats requests will return a partial result if the request encounters any errors. Errors that would have previously terminated the request are now included in the response. [#109008][#109008]

> - [`BACKUP`](../v23.1/backup.html) now skips contacting the ranges for tables on which `exclude_data_from_backup` is set, and can thus succeed even if an excluded table is unavailable. [#109123][#109123]

Author: Fixed thanks!

:kathryn-backups-and-cdc-laser-eyes:

- The default [gRPC]({% link v23.1/architecture/distribution-layer.md %}#grpc) server-side send timeout has been increased from 2 seconds to 4 seconds (1x to 2x of `COCKROACH_NETWORK_TIMEOUT`), to avoid spurious connection failures in certain scenarios. This can be controlled via the new environment variable `COCKROACH_RPC_SERVER_TIMEOUT`. [#109620][#109620]
- Added a new gauge [metric](../v23.1/metrics.html) `sql.schema.invalid_objects`. This gauge is periodically updated based on the schedule set by the `sql.schema.telemetry.recurrence` [cluster setting](../v23.1/cluster-settings.html). When the metric is updated, it counts the number of schema objects ([tables](../v23.1/create-table.html), [types](../v23.1/create-type.html), [schemas](../v23.1/create-schema.html), [databases](../v23.1/create-database.html), and [functions](../v23.1/user-defined-functions.html)) that are in an invalid state according to CockroachDB's internal validation checks. This metric is expected to be zero (`0`) in a healthy cluster. If it is not zero, it indicates that there is a problem that must be repaired. [#109733][#109733]

> - Added two new [changefeed metrics](../v23.1/monitor-and-debug-changefeeds.html#recommended-changefeed-metrics-to-track): `changefeed.checkpoint_progress`  is similar to `changefeed.max_behind_nanos` but supports [metrics labels](../v23.1/monitor-and-debug-changefeeds.html#using-changefeed-metrics-labels); `changefeed.aggregator_progress` tracks the progress of individual aggregators (the lowest timestamp for which all aggregators with the label have emitted all values they're responsible for). [#109744][#109744]

Author: Fixed thanks!

<h3 id="v23-1-11-command-line-changes">Command-line changes</h3>

> - Removed the command `\demo recommission` from [`cockroach demo`]({% link v23.1/cockroach-demo.md %}). It had been obsolete and non-functional ever since v20.2. [#108631][#108631]

Author: Fixed

<h3 id="v23-1-11-db-console-changes">DB Console changes</h3>

> - Users without the `VIEWCLUSTERSETTINGS` [permission]({% link v23.1/security-reference/authorization.md %}#supported-privileges) but with the `VIEWACTIVITY` or `VIEWACTIVITYREDACTED` permissions can now see [index recommendations]({% link v23.1/ui-databases-page.md %}#index-recommendations). [#109466][#109466]

Author: Fixed

- Users with the `VIEWACTIVITY` [privilege]({% link v23.1/security-reference/authorization.md %}#supported-privileges) should be able to see other users' sessions from both the [CLI]({% link v23.1/cockroach-sql.md %}) and the [DB Console]({% link v23.1/ui-overview.md %}). [#108571][#108571]
- Fixed errors on the [**Sessions** page](../v23.1/ui-sessions-page.html) in the [DB Console](../v23.1/ui-overview.html) when a session's memory usage is zero bytes. [#108619][#108619]
- Fixed a bug in [`cockroach demo`]({% link v23.1/cockroach-demo.md %}) whereby `\demo add` could sometimes crash with an error "`index out of range [...] with length ...`". This bug was introduced in v19.*. [#108631][#108631]
- Fixed a bug whereby the command `\demo decommission` in [`cockroach demo`]({% link v23.1/cockroach-demo.md %}) could sometime leave the demo cluster in a broken state. This bug was introduced in v20.2. [#108631][#108631]

> - Fixed a bug in the [index](../v23.1/indexes.html) recommendations provided in the [`EXPLAIN`]({% link v23.1/explain.md %}) output where [`ALTER INDEX ... VISIBLE`](../v23.1/alter-index.html) index recommendations may suggest making the wrong index visible when there are multiple invisible indexes in a table. [cockroachdb/cockroach#108646][#108646]

Reviewer 5: ```suggestion
- Fixed a bug in the [index](../v23.1/indexes.html) recommendations provided in the [`EXPLAIN`]({% link v23.1/explain.md %}) output where [`ALTER INDEX ... VISIBLE`](../v23.1/alter-index.html) index recommendations may suggest making the wrong index visible when there are multiple invisible indexes in a table. [#108646][#108646]
```

Author: Went and fixed all of these, thank you

- Fixed a bug present since v23.1.0 that would cause queries on the `pg_catalog.pg_statistic_ext` table to fail if a table was dropped recently. This bug also caused the `\d` [CLI]({% link v23.1/cockroach-sql.md %}) shortcut to encounter errors. [#108909][#108909]
- Fixed a bug where `pg_attribute` and `pg_attrdef` did not properly return results for generated columns. [#109035][#109035]
- The [Schema Insights](../v23.1/ui-insights-page.html#schema-insights-tab) view should hit request timeouts less frequently, if at all. [#109014][#109014]
- Fixed a bug that caused nodes to crash when attempting to `EXECUTE` a prepared statement with an argument that referenced a [user-defined function]({% link v23.1/user-defined-functions.md %}). This bug was present since user-defined functions were introduced in v22.2. [#108469][#108469]
- Fixed a bug so the filter on the [Statements Page](../v23.1/ui-statements-page.html) works when the app name is an empty string (represented as 'unset'). [#108985][#108985]

> - Fixed a bug where a `SpanStatsRequest` would return post-replicated MVCC stats, which was causing incorrect output in [`SHOW RANGES ... WITH DETAILS`]({% link v23.1/show-ranges.md %}). Now, a `SpanStatsRequest` returns the logical MVCC stats for the requested span. [cockroachdb/cockroach#109234][#109234]

Author: Fixed

> - Fixed an issue with [the "full scan" filter in the UI]({% link v23.1/ui-sql-dashboard.md %}#full-table-index-scans), where the filter was not returning any results. [cockroachdb/cockroach#109274][#109274]

Author: Fixed

- Going to the [Transaction Details Page](https://www.cockroachlabs.com/docs/v23.1/ui-transactions-page#transaction-details-page) from [Workload Insights > High Contention](https://www.cockroachlabs.com/docs/v23.1/ui-insights-page#high-contention) is fixed. Previously, the link would not show any results. [#109253][#109253]

> - Fixed the column names on the [`SELECT` queries]({% link v23.1/selection-queries.md %}) against the tables `crdb_internal.node_txn_execution_insights` and `crdb_internal.cluster_txn_execution_insights` during the [creation of debug.zip files]({% link v23.1/cockroach-debug-zip.md %}). [cockroachdb/cockroach#109515][#109515]

Author: Fixed

- Fixed an issue where a [split](../v23.1/architecture/distribution-layer.html#range-splits) can be called on an invalid key that's in the form of `someValidKey.Next()` during [restore](../v23.1/restore.html). This split key can land in the middle of a row with [column families](../v23.1/column-families.html), and thus result in failing SQL queries when querying the restored table. [#109777][#109777]
- Fixed an issue where a [split](../v23.1/architecture/distribution-layer.html#range-splits) can be called on an invalid key that's in the form of `someValidKey.Next()` during [restore](../v23.1/restore.html) with the `bulkio.restore.use_simple_import_spans` [cluster setting]({% link v23.1/cluster-settings.md %}) set to `true`. This split key can land in the middle of a row with [column families](../v23.1/column-families.html), and thus result in failing SQL queries when querying the restored table. [#109940][#109940]

> - The `difference` [built-in function]({% link v23.1/functions-and-operators.md %}) had its return type incorrectly set to [`string`]({% link v23.1/string.md %}) instead of [`integer`]({% link v23.1/int.md %}). [cockroachdb/cockroach#109752][#109752]

Reviewer 5: ```suggestion
- The `difference` [built-in function]({% link v23.1/functions-and-operators.md %}) had its return type incorrectly set to [`string`]({% link v23.1/string.md %}) instead of [`integer`]({% link v23.1/int.md %}). [#109752][#109752]
```

Author: Fixed

- Fixed a bug where dependencies on [sequences]({% link v23.1/show-sequences.md %}) from [tables]({% link v23.1/show-tables.md %}) would be reported with the wrong value for the `classid` column in the [`pg_catalog.pg_depend` table]({% link v23.1/pg-catalog.md %}). [#110206][#110206]
- Fixed a bug that could cause a [transaction](../v23.1/transactions.html) performing multiple parallel [foreign key]({% link v23.1/foreign-key.md %}) checks to return a `concurrent txn use detected` error. [#110273][#110273]

> - Various observability pages in the [DB Console]({% link v23.1/ui-overview.md %}) no longer crash when they encounter zeros (e.g., a session with no memory allocated). [cockroachdb/cockroach#108785][#108785]

Reviewer 5: ```suggestion
- Various observability pages in the [DB Console]({% link v23.1/ui-overview.md %}) no longer crash when they encounter zeros (e.g., a session with no memory allocated). [#108785][#108785]
```

Author: Fixed
